servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```
@@ -796,6 +820,11 @@ private void removeBlockInternal(long userId, long blockId) throws InvalidStateE
         mMetadataReadLock.unlock();
       }
 
+      if (!location.equals(blockMeta.getBlockLocation())
+          && !location.equals(BlockStoreLocation.anyTier())) {
+        LOG.info("Failed to remove block, because the location is changed!");
```

It might be better to throw an exception than logging the failure.